MetaMask/metamask-extension
# feat: add simulation metrics when simulation UI is not visible

## **Description**

This PR adds transaction simulation metrics when the UI is not shown and the transaction simulation settings is turned on

[![Open in GitHub Codespaces](https://github.com/codespaces/badge.svg)](https://codespaces.new/MetaMask/metamask-extension/pull/28427?quickstart=1)

## **Related issues**

Fixes: https://github.com/MetaMask/metamask-extension/issues/28369

## **Manual testing steps**
Turn on MetaMetrics settings

1. Go to wallet dashboard
2. Initiate send transaction directly through wallet
3. Reject or Cancel the transaction
4. Observe simulation metrics

## **Screenshots/Recordings**



### **Before**



### **After**



## **Pre-merge author checklist**


## **Pre-merge reviewer checklist**
